### DISCUSSION TITLE
Percentage of Purely Black PIxels

### DISCUSSION THREAD

**Discussion by blue-j** [ImageMagick/ImageMagick#5662] [Help] [answered]
I have an odd task to handle - I need to calculate the percentage of pixels that are not exactly 0,0,0 in an RGB image.  Can anyone point me in the right direction?  Much appreciation in advance for your time and knowledge.  : )

- J

**Comment by dlemstra** [comment]
You could use the histogram output of `magick image.png -format %c histogram:info:` to do that but it will require some scripting.

**Comment by snibgo** [comment, marked as answer]
I suggest:
```
magick in.ext -fill White +opaque Black -format %[fx:mean*100] info:

86.4301
```
In my image, 86% of pixels are not black.

EDIT: This assumes the input is fully opaque. If there is transparency, then you need to define what you mean by "black" and "not black".
